### PR TITLE
[-] MO ganalytics: id_product missing in product page

### PR DESCRIPTION
**Pull request by kermes**
sometimes referece is not set and id_product is not set either

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
You could have send it on your previous PR #12 when it was still opened :neutral_face: 

Anyway, I guess we should avoid ternary test in another one.

**Comment by kermes** [comment]
I know.... Just found it right now by chance with a product with no
reference....
Maybe it is better to force id_product im hookProductFooter to avoid more
ternary tests as you suggested. And maybe we should also check of 'id' is
not set either...

File: ganalytics.php [hunk 1]
```diff
@@ -507,7 +507,9 @@ public function hookProductFooter($params)
 		if ($controller_name == 'product')
 		{
 			// Add product view
-			$ga_product = $this->wrapProduct((array)$params['product'], null, 0, true);
+			$product = (array)$params['product'];
+			$product['id_product'] = $product['id'];
```
Comment (Quetzacoalt91) [inline ganalytics.php]: Did you mean ```$params['id']``` instead of ```$product['id']``` ?

**Comment by kermes** [comment]
Yes! I'm sorry, my bad

**Comment by Quetzacoalt91** [comment]
You are right, the ID was on the bottom of my attributes list.

So if we have the ID too, I wonder if we should not do the test directly inside ```wrapProduct(...)``` with something like this:
```php
	public function wrapProduct($product, $extras, $index = 0, $full = false)
	{
		[...]

		$product_id = 0;
		if (!empty($product['reference']))
			$product_id = $product['reference'];
		else if (!empty($product['id_product']))
			$product_id = $product['id_product'];
		else if (!empty($product['id']))
			$product_id = $product['id'];

		[...]

			$ga_product = array(
				'id' => $product_id,
```

The fix could be available for every call of wrapProduct.

**Comment by kermes** [comment]
All right. Right now it works becuase the only place where ['id_product'] is not set is in hookproductfooter, but we can do it as you suggest to avoid future errors.
I'll change it right now.

**Comment by Quetzacoalt91** [comment]
Thanks a lot, I will merge the PR on the dev branch after your commit. Let me know when it is done. :)

**Comment by kermes** [comment]
Done!

**Comment by Quetzacoalt91** [comment]
Merged on the dev branch for the next version.

**Comment by Quetzacoalt91** [comment]
Hi @kermes,

This PR has been merged on the branch master and will be available tomorrow.

Best regards

**Pull request merged**